### ISSUE TITLE
TypeReference(packageName, className) ERRROR

### ISSUE THREAD

**Issue by ELHAMLAOUI** [eclipse/xtext-core#161]
Hi ,
When I try to launch the generation of Xtext artifacts I get the following error:
`[ERROR]: GeneratorException: (Element: -UNKNOWN-; Reported by: XtextGenerator)
     java.lang.IllegalArgumentException: Cannot determine the package name of 'GS.GSPackage'. Please use the TypeReference(packageName, className) constructor`

**Comment by cdietrich** [comment]
see https://www.eclipse.org/forums/index.php/t/1082021/

**Comment by svenefftinge** [comment]
We should add a proper validation for the disallowed naming.

**Comment by cdietrich** [comment]
i had a look and i do not really see a clean solution for that.
what might work is something like

```
public interface IValidationJobCallback {
	
	public boolean isReadOnly();

}
```

in XtextDocumentProvider

```
protected void registerAnnotationInfoProcessor(ElementInfo info) {
		XtextDocument doc = (XtextDocument) info.fDocument;
		if(info.fModel != null) {
			AnnotationIssueProcessor annotationIssueProcessor = new AnnotationIssueProcessor(doc, info.fModel,
				issueResolutionProvider);
			ValidationJob job = new ValidationJob(resourceValidator, doc, annotationIssueProcessor, CheckMode.FAST_ONLY);
			job.setValidationJobCallback(new IValidationJobCallback() {
				
				@Override
				public boolean isReadOnly() {
					return info.fElement != null && XtextDocumentProvider.this.isReadOnly(info.fElement);
					
				}
			});
			doc.setValidationJob(job);
		}
	}
```

```
	@Override
	protected void disposeElementInfo(Object element, ElementInfo info) {
		if (info.fDocument instanceof XtextDocument) {
			XtextDocument document = (XtextDocument) info.fDocument;
			ValidationJob job = (ValidationJob) document.getValidationJob();
			if (job != null) {
				job.cancel();
				job.setValidationJobCallback(null);
			}
			document.disposeInput();
		}
		super.disposeElementInfo(element, info);
	}
```

in ValidationJob

```
	@Override
	protected IStatus run(final IProgressMonitor monitor) {
		if (monitor.isCanceled())
			return Status.CANCEL_STATUS;
		if (callback !=null && callback.isReadOnly()) 
			return Status.OK_STATUS;
                 .....
```

```
	private IValidationJobCallback callback;
        public void setValidationJobCallback(IValidationJobCallback callback) {
		this.callback = callback;
	}
```

@svenefftinge what do you think? do you have a tipp for me?

**Comment by cdietrich** [comment]
i think we can adopt the implementation to retrieve package and interface names separately.
then we do not have any need to do the magic

**Comment by cdietrich** [comment]
fixed in 2.13
